Module 2 Exercises.ipynb

```diff
@@ -1,415 +1,1093 @@
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "      isced11     age    geo  year  frac   income        pop\n",
+       "92072     NaN     NaN   AL01  2019   NaN      NaN   813758.0\n",
+       "92073     NaN     NaN  AL011  2019   NaN      NaN   118948.0\n",
+       "92074     NaN     NaN  AL012  2019   NaN      NaN   290126.0\n",
+       "92075     NaN     NaN  AL013  2019   NaN      NaN    76594.0\n",
+       "92076     NaN     NaN  AL014  2019   NaN      NaN   125195.0\n",
+       "92077     NaN     NaN  AL015  2019   NaN      NaN   202895.0\n",
+       "92078     NaN     NaN   AL02  2019   NaN      NaN  1170142.0\n",
+       "92079     NaN     NaN  AL021  2019   NaN      NaN   274982.0\n",
+       "92080     NaN     NaN  AL022  2019   NaN      NaN   895160.0\n",
+       "92081     NaN     NaN   AL03  2019   NaN      NaN   878527.0\n",
+       "92082     NaN     NaN  AL031  2019   NaN      NaN   125157.0\n",
+       "92083     NaN     NaN  AL032  2019   NaN      NaN   294747.0\n",
+       "92084     NaN     NaN  AL033  2019   NaN      NaN    61423.0\n",
+       "92085     NaN     NaN  AL034  2019   NaN      NaN   207889.0\n",
+       "92086     NaN     NaN  AL035  2019   NaN      NaN   189311.0\n",
+       "92098   ED5-8  Y20-24   AT11  2019  35.8  24300.0   293433.0\n",
+       "92099   ED5-8  Y25-34   AT11  2019  39.0  24300.0   293433.0\n",
+       "92100   ED5-8  Y25-64   AT11  2019  30.7  24300.0   293433.0\n",
+       "92101   ED5-8  Y30-34   AT11  2019  41.7  24300.0   293433.0\n",
+       "92102     NaN     NaN  AT111  2019   NaN      NaN    37513.0\n",
+       "92103     NaN     NaN  AT112  2019   NaN      NaN   158981.0\n",
+       "92104     NaN     NaN  AT113  2019   NaN      NaN    96939.0\n",
+       "92105   ED5-8  Y20-24   AT12  2019  31.6  25900.0  1677542.0\n",
+       "92106   ED5-8  Y25-34   AT12  2019  38.9  25900.0  1677542.0\n",
+       "92107   ED5-8  Y25-64   AT12  2019  32.4  25900.0  1677542.0\n",
+       "92108   ED5-8  Y30-34   AT12  2019  39.1  25900.0  1677542.0\n",
+       "92109     NaN     NaN  AT121  2019   NaN      NaN   246740.0\n",
+       "92110     NaN     NaN  AT122  2019   NaN      NaN   260480.0\n",
+       "92111     NaN     NaN  AT123  2019   NaN      NaN   155446.0\n",
+       "92112     NaN     NaN  AT124  2019   NaN      NaN   217445.0"
       ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df1 = estat_edat_lfse_04_tidy.loc[estat_edat_lfse_04_tidy['age'] == \"Y25=64\"]\n",
+    "df1 = estat_edat_lfse_04_tidy.loc[estat_edat_lfse_04_tidy['isced11'] == \"ED5-8\"]\n",
+    "df2 = estat_nama_10r_2hhinc_tidy\n",
+    "df3 = estat_demo_r_pjanaggr3_tidy.loc[estat_demo_r_pjanaggr3_tidy['age'] == \"TOTAL\"].drop(columns=[\"age\"])\n",
+    "df = pd.merge(df1, df2, on=[\"year\", \"geo\"], how='outer').merge(df3, on=[\"year\", \"geo\"], how = \"outer\")\n",
```

Comment:
For readability, you may consider breaking these lines into multiple lines. E.g.:

```
df3 = (
    estat_demo_r_pjanaggr3_tidy
    .loc[estat_demo_r_pjanaggr3_tidy['age'] == "TOTAL"]
    .drop(columns=["age"])
)
```